libs/utils/android/screen.py

```diff
@@ -155,4 +157,70 @@ def unlock(target):
        System.menu(target)
        System.home(target)
 
+    class Recorder(object):
+        """
+        Android screenrecord utility class
+        """
+
+        def __init__(self, target):
+            self._target = target
+            self._log = logging.getLogger('Recorder')
+
+            self._recording = False
+
+        def start(self, out_file,
+                         timestamp=True, max_time=None):
+            """
+            Start recording the screen of the device
+            """
+            if self._recording:
+                raise RuntimeError('Recording already started')
+
+            self._local_file = out_file
+            self._remote_file = self._target.path.join(
+                self._target.working_directory,
+                os.path.basename(out_file)
+            )
+
+            self._max_time = max_time
+
+            options = []
+            if timestamp:
+                options.append('--bugreport')
+            if max_time:
+                options.append('--time-limit {}'.format(max_time))
+
+            record_cmd = 'screenrecord {} {}'.format(
+                ' '.join(options), self._remote_file
+            )
+
+            self._log.debug('screenrecord command = \"{}\"'.format(record_cmd))
+            self._log.info('Starting screen recording')
+
+            self._recording = True
+            self._process = self._target.background(record_cmd, as_root=True)
+
+        def stop(self):
+            """
+            Stop recording the screen of the device
+            """
+            if not self._recording:
+                raise RuntimeError('Recording not started')
+
+            if not self._max_time:
+                # Recording must be ended by Ctrl-C on the device
+                self._log.debug('Sending Ctrl-C to screenrecord')
+                self._target.execute('kill -INT $(pgrep screenrecord)', as_root=True)
+
+            self._log.info('Waiting for recording to be completed')
+            self._process.wait()
```

Comment:
If I understand correctly we should really wait here only if `max_time` was specified is that correct ? If yes, I think `stop()` should always end the record immediately and the wait semantics should be provided by another method. That way, a user could do `start() ... wait()` to record the screen for certain amount of time or `start() ... stop()` in an event-based fashion with `max_time` just serving as a timeout.